kpatch-build/kpatch-build

```diff
@@ -1026,10 +1084,22 @@ if [[ -z "$OOT_MODULE" ]]; then
 else
 	KPATCH_BUILD="/lib/modules/$ARCHVERSION/build"
 fi
+
+MAKEVARS=()
+if [ "$CONFIG_CC_IS_CLANG" -eq 1 ]; then
+	# We don't need kpatch-cc for module build, so re-export CC and LD
+	# variables with it removed
+
+	MAKEVARS+=("CC=\"clang\"")
+	if [ -n "$LD" ]; then
+		MAKEVARS+=("LD=\"$LD\"")
+	fi
+fi
```

Comment:
> I'm not sure these variables are even getting passed to the module build. This is using kmod/patch/Makefile which doesn't check for these variables or even pass them along.

They remain set when we call $(MAKE) -C $(KPATCH_BUILD) M=$(PWD)

> But then again, I don't think it's even necessary for the makefile to do so, because out-of-tree module builds are already supposed to use the compiler listed in the .config.
> So maybe MAKEVARS isn't even needed here?

Tried removing MAKEVARS here and the build failed because it tried to build with gcc as a compiler but with clang CFLAGS. I looked through kbuild and CC is never set based on CONFIG_CC_IS_CLANG, it is the other way around.